tests/integration/targets/elb_application_lb/tasks/main.yml

```diff
@@ -1,12 +1,1137 @@
+          - elb is not changed
+          - elb.ip_address_type == 'ipv4'
+
+    # ------------------------------------------------------------------------------------------
+
+    - name: Update an ELB with different listener by adding rule - check_mode
+      elb_application_lb:
+        name: "{{ elb_name }}"
+        subnets: "{{ public_subnets }}"
+        security_groups: "{{ sec_group.group_id }}"
+        state: present
+        listeners:
+          - Protocol: HTTP
+            Port: 80
+            DefaultActions:
+              - Type: forward
+                TargetGroupName: "{{ tg_name }}"
+            Rules:
+              - Conditions:
+                  - Field: path-pattern
+                    Values:
+                      - '/test'
+                Priority: '1'
+                Actions:
+                  - TargetGroupName: "{{ tg_name }}"
+                    Type: forward
+      register: elb
+      check_mode: yes
+
+    - assert:
+        that:
+          - elb is changed
+          - elb.msg is match('Would have updated ELB if not in check mode.')
+
+    - name: Update an ELB with different listener by adding rule
+      elb_application_lb:
+        name: "{{ elb_name }}"
+        subnets: "{{ public_subnets }}"
+        security_groups: "{{ sec_group.group_id }}"
+        state: present
+        listeners:
+          - Protocol: HTTP
+            Port: 80
+            DefaultActions:
+              - Type: forward
+                TargetGroupName: "{{ tg_name }}"
+            Rules:
+              - Conditions:
+                  - Field: path-pattern
+                    Values:
+                      - '/test'
+                Priority: '1'
+                Actions:
+                  - TargetGroupName: "{{ tg_name }}"
+                    Type: forward
+      register: elb
+
+    - assert:
+        that:
+          - elb is changed
+          - elb.listeners[0].rules | length == 2
+          - "'1' in {{ elb.listeners[0] | community.general.json_query('rules[*].priority') }}"
+
+    - name: Update an ELB with different listener by adding rule (idempotence) - check_mode
+      elb_application_lb:
+        name: "{{ elb_name }}"
+        subnets: "{{ public_subnets }}"
+        security_groups: "{{ sec_group.group_id }}"
+        state: present
+        listeners:
+          - Protocol: HTTP
+            Port: 80
+            DefaultActions:
+              - Type: forward
+                TargetGroupName: "{{ tg_name }}"
+            Rules:
+              - Conditions:
+                  - Field: path-pattern
+                    Values:
+                      - '/test'
+                Priority: '1'
+                Actions:
+                  - TargetGroupName: "{{ tg_name }}"
+                    Type: forward
+      register: elb
+      check_mode: yes
+
+    - assert:
+        that:
+          - elb is not changed
+          - elb.msg is match('IN CHECK MODE - no changes to make to ELB specified.')
```

Comment:
@jillr Do we want to keep all this messaging for the elb? I guess this comes from module_utils.